api/src/test/java/org/openmrs/obs/AbstractHandlerTest.java

```diff
@@ -42,119 +32,75 @@ public class AbstractHandlerTest extends BaseContextSensitiveTest {
-		handler = new  AbstractHandler();
+	public void initializeContext() throws APIException {
+		handler = new  AbstractHandler(adminService, storageService);
+		
 		adminService.saveGlobalProperty(new GlobalProperty(
-			OpenmrsConstants.GLOBAL_PROPERTY_COMPLEX_OBS_DIR,
-			complexObsTestFolder.toAbsolutePath().toString()
+			OpenmrsConstants.GLOBAL_PROPERTY_COMPLEX_OBS_DIR, "obs"
 		));
 	}
 	
 	@Test
-	public void getOutputFileToWrite_shouldNeverOverwritePreviousFiles() throws IOException {
+	public void saveObs_shouldNeverOverwritePreviousFiles() {
 		String content1 = "A";
 		String content2 = "B";
 		
-		File previousFile = null;
-		File currentFile = null;
-		
 		for (int i = 0; i <= 101; i++) {
 			String currentData = (i % 2 == 0) ? content1 : content2;
 			
-			ComplexData complexData = new ComplexData(FILENAME, currentData);
+			ComplexData complexData = new ComplexData(FILENAME, currentData.getBytes(StandardCharsets.UTF_8));
 			
 			Obs obs = new Obs();
 			obs.setComplexData(complexData);
 			
-			currentFile = handler.getOutputFileToWrite(obs);
-			
-			try (BufferedWriter fout = new BufferedWriter(
-			        new OutputStreamWriter(new FileOutputStream(currentFile), StandardCharsets.UTF_8))) {
-				fout.write(currentData);
-			}
-			
-			try (BufferedReader fin = new BufferedReader(
-			        new InputStreamReader(new FileInputStream(currentFile), StandardCharsets.UTF_8))) {
-				String readData = fin.readLine();
-				assertEquals(readData, currentData);
-			}
-			
-			if (i > 0) {
-				assertFalse(FileUtils.contentEquals(previousFile, currentFile));
-			}
+			handler.saveObs(obs);
+
+			Obs fetchedObs = handler.getObs(obs, null);
 			
-			previousFile = currentFile;
+			assertEquals(currentData, new String((byte[]) fetchedObs.getComplexData().getData()));
 		}
 	}
 	
 	@Test
-	public void getOutputFileToWrite_shouldCorrectlyNameTitledFileWithExtension() throws IOException, ParseException {
-		ComplexData complexDataWithTitle = new ComplexData(FILENAME, null);
+	public void saveObs_shouldPreserveTitleWithoutExtension() {
+		ComplexData complexDataWithTitle = new ComplexData(FILENAME, "A".getBytes(StandardCharsets.UTF_8));
 		
 		Obs obsWithTitle = new Obs();
 		obsWithTitle.setComplexData(complexDataWithTitle);
 		
-		File titledFile = handler.getOutputFileToWrite(obsWithTitle);
-		titledFile.createNewFile();
+		handler.saveObs(obsWithTitle);
 		
-		String[] nameWithTitle = titledFile.getName().split("_|\\.");
+		String[] nameWithTitle = obsWithTitle.getValueComplex().split("_|\\.");
 		
 		String titlePart = nameWithTitle[0];
-		String uuidPartWithTitle = nameWithTitle[1];
-		String extensionPart = nameWithTitle[2];
 		
 		assertEquals(titlePart, FilenameUtils.removeExtension(FILENAME));
-		assertEquals(extensionPart, "txt");
-		assertEquals(uuidPartWithTitle, obsWithTitle.getUuid());
-	}
-	
-	@Test
-	public void getOutputFileToWrite_shouldCorrectlyNameTitledFileWithoutExtension() throws IOException, ParseException {
```

Comment:
Are we removing this because we changed the file naming convention?

Reply:
Correct. The outputfile is also no longer available as a file handle, just inputstream.